packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -543,19 +549,11 @@ export class ContextualMenuBase extends React.Component<IContextualMenuProps, IC
         primaryDisabled: item.primaryDisabled,
       };
 
-      const menuItemStyles = this._classNames.subComponentStyles
-        ? (this._classNames.subComponentStyles.menuItem as IStyleFunctionOrObject<
-            IContextualMenuItemStyleProps,
-            IContextualMenuItemStyles
-          >)
-        : undefined;
-
       // We need to generate default styles then override if styles are provided
       // since the ContextualMenu currently handles item classNames.
-      itemClassNames = mergeStyleSets(
-        getContextualMenuItemClassNames(getItemStyles, itemStyleProps),
-        getContextualMenuItemClassNames(menuItemStyles, itemStyleProps),
-        getContextualMenuItemClassNames(styles, itemStyleProps),
+      itemClassNames = getContextualMenuItemClassNames(
+        _getMenuItemStylesFunction(this._classNames.subComponentStyles?.menuItem, styles),
```

Comment:
hmm, it would be easy to frequently mutate this no? same with `styles`. Will this save computations or just fill up the memoization cache?

Reply:
so we know we are not mutating `this._classNames.subComponentStyles?.menuItem` and `styles` is passed by user. 

so in case where user passed a mutating styles prop, it's going to get filled up and get bad perf (regardless of my change)
But in case user did the right thing, they going to get much better perf than before. 

my change should be only making things better but not worse :)